deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/FileSystemDeploymentService.java

```diff
@@ -1277,19 +1284,28 @@ private void writeFailedMarker(final File deploymentFile, final String failureDe
         }
     }
 
-    private static File[] listDirectoryChildren(File directory) {
-        return listDirectoryChildren(directory, null);
+    private static List<File> listDirectoryChildren(File directory) {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(directory.toPath())) {
+            final List<File> result = new ArrayList<>();
+            stream.forEach(entry -> result.add(entry.toFile()));
+            return result;
+        } catch (SecurityException | IOException ex) {
+            throw DeploymentScannerLogger.ROOT_LOGGER.cannotListDirectoryFiles(ex, directory);
+        }
     }
 
-    private static File[] listDirectoryChildren(File directory, FileFilter filter) {
-        File[] result = directory.listFiles(filter);
-        if (result == null) {
-            throw DeploymentScannerLogger.ROOT_LOGGER.cannotListDirectoryFiles(directory);
+    private static List<File> listDirectoryChildren(File directory, DirectoryStream.Filter<Path> filter) {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(directory.toPath(), filter)) {
+            final List<File> result = new ArrayList<>();
+            stream.forEach(entry -> result.add(entry.toFile()));
```

Comment:
The goal was to get more informations about the error during the file listing 
Going all the way to nio with its quirks is not for this pr tuo make ;)